### PR TITLE
[Rust SDK] multi_get_object change to not fail on one fail

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-json-rpc/src/read_api.rs [hunk 1]
```diff
@@ -155,20 +155,18 @@ impl ReadApiServer for ReadApi {
...
+            let objects: Vec<SuiObjectResponse> = results
+                .into_iter()
+                .map(|result| match result {
+                    Ok(response) => Ok(response),
+                    Err(error) => {
+                        error!("Failed to fetch object with error: {error:?}");
+                        Err(format!("Error: {}", error))
+                    }
+                })
+                .filter_map(Result::ok)
```
Comment (666lcz) [inline crates/sui-json-rpc/src/read_api.rs:167]: We should not filter out the error. We can update SuiObjectResponseError::unknow to take in a string, or even better, add a new DisplayError variant

Comment (666lcz) [reply crates/sui-json-rpc/src/read_api.rs:167]: Should we also update `get_owned_objects`? Can you also double check the indexer native implementation?

Comment (healthydeve) [reply crates/sui-json-rpc/src/read_api.rs:167]: `get_owned_objects` got moved to an indexer only native impl by @patrickkuo

Comment (666lcz) [reply crates/sui-json-rpc/src/read_api.rs:167]: No, it still exists on fullnode: https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc/src/indexer_api.rs#L67

**Comment by healthydeve** [comment]
Closing this in favor of the branch on main

**Pull request closed without merging**